File: 백준/Silver/9020. 골드바흐의 추측/골드바흐의 추측.java
import java.util.*;
import java.io.*;
public class Main {
    public static boolean isPrime(int x) {
        if (x == 1) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(x); i++) {
            if (x % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        for (int i = 0; i < T; i++) {
            int N = Integer.parseInt(br.readLine());

            int a = N / 2;
            int b = N / 2;

            while (a > 0) {
                if (isPrime(a) && isPrime(b)) {
                    System.out.println(a + " " + b);
                    break;
                } else {
                    a -= 1;
                    b += 1;
                }
            }

        }

    }
}

File: 백준/Silver/9020. 골드바흐의 추측/골드바흐의 추측.py
from math import sqrt
import sys
input = sys.stdin.readline

def isPrime(x):
    if x == 1:
        return False
    for k in range(2, int(sqrt(x)) + 1):
        if x % k == 0:
            return False
    return True

T = int(input())
for _ in range(T):
    num = int(input())
    a, b = num // 2, num // 2

    while a > 0:
        if isPrime(a) and isPrime(b):
            print(a, b)
            break
        else:
            a -= 1
            b += 1
